Allow edit of priority field on EditCommand

LeongWZ opened the pull request:
Resolves #46. 

This commit allows user to edit the priority field on `Person` class.
The user may edit the priority field by typing `edit INDEX pri/ priority_value"`, e.g. `edit 1 pri/ HIGH`.

LeongWZ committed:
Allow edit of priority field on EditCommand

Gra7ityIC3 commented on src/test/java/seedu/address/logic/commands/EditCommandTest.java:
```diff
@@ -71,43 +75,20 @@ public void execute_someFieldsSpecifiedUnfilteredList_success() {
+        helper.execute_specifiedRemarkField_shouldModifyRemark("");
+    }
+
+    @Test void execute_specifiedPriorityField_shouldModifyPriority() {
```
Could you specify the access modifier and place the `@Test` annotation above the method signature?

LeongWZ replied:
Noted. Thanks for catching that.

Gra7ityIC3 reviewed (commented):
Aside from some minor changes, LGTM.

LeongWZ committed:
Specify access modifier in test method

Specified `excecute_specifiedPriorityField_shouldModifyPriority`
method's access modifier as public. In addition, private `Helper` class
in `EditCommandTest` was also refactored to improve code quality; created new
`generateExpectedModel` and `generateExpectedSuccessMEssage` inline
methods.

LeongWZ committed:
Delete Age class

TerenceWongsExtravaganza commented on src/test/java/seedu/address/logic/commands/EditCommandTest.java:
```diff
@@ -255,13 +254,22 @@ public void execute_specifiedPriorityField_shouldModifyPriority(Priority priorit
         }
     }
+
+    private Model generateExpectedModel(Person personToBeEdited, Person personToBeExpected) {
```
Good job abstracting out the model creation process out of testcase!

Gra7ityIC3 reviewed (approved):
LGTM!
